src/Azure/Queues/AzureServiceBusQueue.cs

```diff
@@ -21,22 +21,33 @@ public class AzureServiceBusQueue<T> : QueueBase<T> where T : class {
         private long _workerErrorCount;
         private readonly int _retries;
         private readonly TimeSpan _workItemTimeout = TimeSpan.FromMinutes(5);
+        private readonly TimeSpan _autoDeleteOnIdle = TimeSpan.MaxValue;
+        private readonly TimeSpan _defaultMessageTimeToLive = TimeSpan.MaxValue;
 
-        public AzureServiceBusQueue(string connectionString, string queueName = null, int retries = 2, TimeSpan? workItemTimeout = null, bool shouldRecreate = false, RetryPolicy retryPolicy = null, ISerializer serializer = null, IEnumerable<IQueueBehavior<T>> behaviors = null, ILoggerFactory loggerFactory = null) : base(serializer, behaviors, loggerFactory) {
+        public AzureServiceBusQueue(string connectionString, string queueName = null, int retries = 2, TimeSpan? workItemTimeout = null, bool shouldRecreate = false, RetryPolicy retryPolicy = null, ISerializer serializer = null, IEnumerable<IQueueBehavior<T>> behaviors = null, ILoggerFactory loggerFactory = null, TimeSpan? autoDeleteOnIdle = null, TimeSpan? defaultMessageTimeToLive = null) : base(serializer, behaviors, loggerFactory) {
             _queueName = queueName ?? typeof(T).Name;
             _namespaceManager = NamespaceManager.CreateFromConnectionString(connectionString);
             _retries = retries;
             if (workItemTimeout.HasValue && workItemTimeout.Value < TimeSpan.FromMinutes(5))
                 _workItemTimeout = workItemTimeout.Value;
 
+            if (autoDeleteOnIdle.HasValue && autoDeleteOnIdle.Value >= TimeSpan.FromMinutes(5))
+                _autoDeleteOnIdle = autoDeleteOnIdle.Value;

```

Comment:
That might be a requirement of Azure like `workItemTimeout.Value < TimeSpan.FromMinutes(5)` @trbenning?